src/hooks.js

```diff
@@ -0,0 +1,50 @@
+import { useEffect, useState } from 'react';
+import { getAuthenticatedHttpClient } from '@edx/frontend-platform/auth';
+import getLearnerPortalLinks from './learnerPortalLinks';
+import { getSelectedEnterpriseUUID } from './utils';
+
+export default function useEnterpriseConfig(
+  authenticatedUser,
+  learnerPortalHostname,
+  lmsBaseUrl,
+) {
+  const [enterpriseLearnerPortalLink, setEnterpriseLearnerPortalLink] = useState();
+  const [enterpriseCustomerBrandingConfig, setEnterpriseCustomerBrandingConfig] = useState();
+  useEffect(
+    () => {
+      const httpClient = getAuthenticatedHttpClient();
+      getLearnerPortalLinks(
+        httpClient,
+        authenticatedUser,
+        learnerPortalHostname,
+        lmsBaseUrl,
+      ).then((learnerPortalLinks) => {
+        const preferredUUID = getSelectedEnterpriseUUID(authenticatedUser);
+        const preferredLearnerPortalLink = learnerPortalLinks.find(learnerPortalLink =>
+          learnerPortalLink.uuid === preferredUUID);
+        if (preferredLearnerPortalLink) {
+          const config = {
+            logoAltText: preferredLearnerPortalLink.title,
+            logoDestination: preferredLearnerPortalLink.url,
+          };
+          if (preferredLearnerPortalLink.branding_configuration) {
+            config.logo = preferredLearnerPortalLink.branding_configuration.logo;
+          }
+          setEnterpriseCustomerBrandingConfig(config);
```

Comment:
If `branding_configuration` is not set, I'm not sure we want/need to call `setEnterpriseCustomerBrandingConfig(config)` since a branding configuration does not exist.

Reply:
I wonder if `setEnterpriseCustomerBrandingConfig` is a bit of a misnomer - `config`, here, has the title and the url of the learner portal, and possibly the logo from the `branding_configuration`.  Maybe we can include the title and the logo (if it exists) in the `enterpriseLearnerPortalLink` state object and deprecate the `enterpriseCustomerBrandingConfig` object, since they're kinda doing the same thing?